flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/exceptionhistory/RootExceptionHistoryEntryTest.java

```diff
@@ -81,10 +81,14 @@ void testFromFailureHandlingResultSnapshot() throws ExecutionException, Interrup
         final CompletableFuture<Map<String, String>> rootFailureLabels =
                 CompletableFuture.completedFuture(Collections.singletonMap("key", "value"));
 
-        final Throwable concurrentException = new IllegalStateException("Expected other failure");
-        final ExecutionVertex concurrentlyFailedExecutionVertex = extractExecutionVertex(1);
-        final long concurrentExceptionTimestamp =
-                triggerFailure(concurrentlyFailedExecutionVertex, concurrentException);
+        final Throwable concurrentException1 = new IllegalStateException("Expected other failure1");
+        final ExecutionVertex concurrentlyFailedExecutionVertex1 = extractExecutionVertex(1);
+        Predicate<ExceptionHistoryEntry> exception1Predicate =
+                getExceptionHistoryEntryPredicate(
+                        concurrentException1, concurrentlyFailedExecutionVertex1);
+
+        final Throwable concurrentException2 = new IllegalStateException("Expected other failure2");
+        final ExecutionVertex concurrentlyFailedExecutionVertex2 = extractExecutionVertex(2);
```

Comment:
Can you move these two lines further down where they are actually used (alternatively, you could move the Predicate initialization up). The goal would be to have the entire initialization in one place. Me personally, I would prefer moving the code down. That gives less distraction to the reader so that he/she can focus on the first concurrent exception handling at the beginning. But up to you...

Reply:
Thinking about it: Another solution would be to create a dedicated test method `testAddConecurrentExceptions`. WDYT? :thinking:

Reply:
My thoughts is moving the code down instead of create a new test method.

Because `testAddConecurrentExceptions` will use all code of this test. WDYT?

Reply:
Fine with me. About reusing the code: You could move the code that's common between tests into a private static method. But other's might argue that this is also not good for code readability.

Anyway, it's your choice in the end. Most of my comments are discussion points rather than comments where I claim that my proposal is the right one. :-)

Reply:
Thanks for the explanation!

> testAddConecurrentExceptions will use all code of this test 

If we have 2 tests, and test1 only call `testCommon`, and test2 call `testCommon` and `testPart2`. It means test2 can cover test1.

2 solutions:
- keep test1 and test2
- Only keep test2

In the current scenario, is it enough for us to only keep test2? Looking forward to your opinion, fine with me as well. 😁

Reply:
Fine with me to keep it like that.